src/goto-instrument/contracts/contracts.cpp

```diff
@@ -636,7 +742,7 @@ void code_contractst::instrument_assign_statement(
     instruction_it->is_assign(),
     "The first instruction of instrument_assign_statement should always be"
     " an assignment");
-
+  add_pragma_disable_assigns_check(*instruction_it);
```

Comment:
Could you add a comment here explaining why do we need to disable inclusion checks at this point? It might make debugging easier in the future.

Reply:
The idea is to not instrument the same instruction again. Do you think it's worth adding a comment on that?

Reply:
I moved this line to the `for` loop that iterated over all instructions, instead of repeating it in all `instrument_*` functions, and added a clarifying comment.

Reply:
> I moved this line to the `for` loop that iterated over all instructions, instead of repeating it in all `instrument_*` functions, and added a clarifying comment.

That's perfect! Thank you.